plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1287,15 +1288,17 @@ private static Location getValidatedExternalLocation(String location)
         return validated;
     }
 
-    private void checkExternalPath(HdfsContext context, Path path)
+    private void checkExternalPathAndCreateIfNotExists(HdfsContext context, Path path)
     {
         try {
-            if (!isS3FileSystem(context, hdfsEnvironment, path)) {
-                if (!hdfsEnvironment.getFileSystem(context, path).isDirectory(path)) {
-                    throw new TrinoException(INVALID_TABLE_PROPERTY, "External location must be a directory: " + path);
-                }
+            FileSystem fileSystem = hdfsEnvironment.getFileSystem(context, path);
+            if (!fileSystem.isDirectory(path)) {
```

Comment:
We used this method before and rely on logic, that in case of FileNotFoundException this method returns false, not to introduce additional try/catch logic